src/tinkerio.py

```diff
@@ -947,22 +956,37 @@ def molecular_dynamics(self, nsteps, timestep, temperature=None, pressure=None,
         if nequil > 0:
             write_key("%s-eq.key" % self.name, eq_opts, "%s.key" % self.name, md_defs)
             if verbose: printcool("Running equilibration dynamics", color=0)
+            if 'OPENMM_CUDA_COMPILER' in os.environ.keys() and 'gas' not in self.name:
+                dynamickeyword='dynamic_omm'
+            else:
+                dynamickeyword='dynamic'
             if self.pbc and pressure is not None:
-                self.calltinker("dynamic %s -k %s-eq %i %f %f 4 %f %f" % (self.name, self.name, nequil, timestep, float(nsave*timestep)/1000, 
-                                                                          temperature, pressure), print_to_screen=verbose)
+                if 'OPENMM_CUDA_COMPILER' in os.environ.keys():
+                    self.calltinker(dynamickeyword+" %s -k %s-eq %i %f %f 4 %f %f N" % (self.name, self.name, nequil, timestep, float(nsave*timestep)/1000,temperature, pressure), print_to_screen=verbose)
+
+
+                else:
+                    self.calltinker(dynamickeyword+" %s -k %s-eq %i %f %f 4 %f %f" % (self.name, self.name, nequil, timestep, float(nsave*timestep)/1000,temperature, pressure), print_to_screen=verbose)
             else:
-                self.calltinker("dynamic %s -k %s-eq %i %f %f 2 %f" % (self.name, self.name, nequil, timestep, float(nsave*timestep)/1000,
-                                                                       temperature), print_to_screen=verbose)
+                self.calltinker(dynamickeyword+" %s -k %s-eq %i %f %f 2 %f" % (self.name, self.name, nequil, timestep, float(nsave*timestep)/1000,temperature), print_to_screen=verbose)
             os.system("rm -f %s.arc" % (self.name))
 
         # Run production.
         if verbose: printcool("Running production dynamics", color=0)
         write_key("%s-md.key" % self.name, md_opts, "%s.key" % self.name, md_defs)
+        if 'OPENMM_CUDA_COMPILER' in os.environ.keys() and 'gas' not in self.name:
+             dynamickeyword='dynamic_omm'
+        else:
+             dynamickeyword='dynamic'
+
         if self.pbc and pressure is not None:
-            odyn = self.calltinker("dynamic %s -k %s-md %i %f %f 4 %f %f" % (self.name, self.name, nsteps, timestep, float(nsave*timestep/1000), 
-                                                                             temperature, pressure), print_to_screen=verbose)
+            if 'OPENMM_CUDA_COMPILER' in os.environ.keys():
```

Comment:
Is there some way to simplify this so that it doesn't check for OPENMM_CUDA_COMPILER twice (once when it determines dynamickeyword and again when it decides how to call TINKER?)  Maybe you could make the N into a variable "dynamicsuffix" or something like that?